app.py

```diff
@@ -21,41 +22,53 @@
     authority=app.config.get("AUTHORITY"),
     client_id=app.config["CLIENT_ID"],
     client_credential=app.config["CLIENT_SECRET"],
-    )
+)
+
 
 @app.route("/login")
 def login():
-    return render_template("login.html", version=identity.__version__, **auth.log_in(
-        scopes=app_config.SCOPE,  # Have user consent scopes during log-in
-        redirect_uri=url_for("auth_response", _external=True),  # Optional. If present, this absolute URL must match your app's redirect_uri registered in Azure Portal
-        ))
+    result = auth.log_in(
+        # Have user consent scopes during log-in
+        scopes=app_config.SCOPE,
+        # Optional. If present, this absolute URL must match your app's redirect_uri registered in Azure Portal
+        redirect_uri=url_for("auth_response", _external=True),
+    )
+    return render_template("login.html", version=identity.__version__, **result)
+
 
 @app.route(app_config.REDIRECT_PATH)
 def auth_response():
     result = auth.complete_log_in(request.args)
-    return render_template("auth_error.html", result=result) if "error" in result else redirect(url_for("index"))
+    if "error" not in result:
```

Comment:
We typically avoid ternary operators since not every developer is comfortable with them, and they can be harder to debug. They also tend to produce excessively long lines.

Reply:
It is fine to avoid ternary operator. But then I would suggest to swap your two branches here into:

```python
if "error" in result:  # <-- We detected an error
    return render_template("auth_error.html", result=result)  # <-- Then we handle that error inside the same if clause
return redirect(url_for("index"))  # <-- This is the happy path
```

Reply:
Good point, swapping